src/lib.rs

```diff
@@ -182,50 +182,69 @@ pub mod mock;
 //
 // can't use both sync and async secret service
 //
-#[cfg(all(feature = "sync-secret-service", feature = "async-secret-service"))]
+#[cfg(any(
+    all(feature = "sync-secret-service", feature = "async-secret-service"),
+    all(
+        feature = "sync-persistent-keyutils",
+        feature = "async-persistent-keyutils",
+    )
+))]
 compile_error!("This crate cannot use the secret-service both synchronously and asynchronously");
 
 //
 // pick the *nix keystore
 //
-
-#[cfg(all(target_os = "linux", feature = "linux-native"))]
+#[cfg(all(target_os = "linux", feature = "keyutils"))]
 pub mod keyutils;
-// use keyutils as default if secret-service is not available
 #[cfg(all(
     target_os = "linux",
     feature = "linux-native",
-    not(any(feature = "sync-secret-service", feature = "async-secret-service"))
+    not(feature = "sync-secret-service"),
+    not(feature = "async-secret-service"),
 ))]
```

Comment:
I like the unrolling of the not-any!! And now I see why you wanted to separate linux-native from keyutils.  But that's not necessary, because if someone has enabled either version of the persistent-keyutils store they will also have enabled one of the secret-service features, so keyutils will still not be chosen as the default.